File: 14-longest-common-prefix/longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        if not strs:
            return ""
        
        pre = strs[0]
        for i in strs[1:]:
            count = 0 
            while(count < len(pre) and count < len(i) and pre[count] == i[count]):
                count += 1
            pre = pre[:count]
            if (pre == ""):
                break
        return pre







        

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String f = strs[0];

        for(String s : strs)
        {
            while(!s.startsWith(f))
            {
                f = f.substring(0,f.length() - 1);
                if(f.isEmpty()) return "";
            }
        }
        return f;
    }
}